tracee-ebpf/tracee/argprinters.go

```diff
@@ -57,47 +57,47 @@ func (t *Tracee) prepareArgs(ctx *context, args map[string]interface{}) error {
 	switch ctx.EventID {
 	case SysEnterEventID, SysExitEventID, CapCapableEventID, CommitCredsEventID, SecurityFileOpenEventID:
 		//show syscall name instead of id
-		if id, isInt32 := args["syscall"].(int32); isInt32 {
+		if id, isInt32 := args["syscall"].(int32); isInt32 && !t.config.Output.RawArguments {
```

Comment:
Instead of adding this `!t.config.Output.RawArguments` for almost every case, I would group all these cases together, and use one `if` check to wrap them all.

This also makes me think about the two places where you didn't add this check:
1. Printing alerts
2. Socket address

Maybe we shuold handle these in a diferent place? e.g. event_decode?

Reply:
Should alerts be disabled if raw is specified? I think it would be the raw struct if not parsed, which I don't think is any more machine readable.

Reply:
No, alerts should be kept as strings. There is no meaning (outside of tracee code) to the alerts enum

Reply:
So what would the check do relating to printing alerts?

Reply:
`So what would the check do relating to printing alerts?`

Not sure I understand what you ask here.
What I am saying is, that instead of checking `!t.config.Output.RawArguments` many times, just do it one time, and place all the relevant arguments (not alerts or socket address arguments) under this check.

Reply:
Ah - I was confused by your comment at the top of this thread. Makes sense now.

Reply:
I made changes where args were parsed in `readArgFromBuf()` so that only one check would have to happen for `t.config.Output.ParseArguments` but honestly I like the way it is in now better. It's clearer in the code to have all the argument types for each EventID checked in one place.

Reply:
Wait, changes in `readArgFromBuf()`?

What I had in mind is something like the following:
switch ctx.EventID {
    case MemProtAlertEventID:
    ....
    // all the cases which are not changed come here (alerts and socket arguments)
}

if !t.config.Output.ParseArguments {
    return nil
}

switch ctx.EventID {
    case SysEnterEventID, SysExitEventID, CapCapableEventID, CommitCredsEventID, SecurityFileOpenEventID:
    ....
    // all other cases (which you changed in this commit) comes here, without any change

Reply:
I thought by `event_decode` you were referring to `events_decoder.go`, regardless miscommunication and I agree with your idea here. Just pushed these changes.

Reply:
> I thought by event_decode you were referring to events_decoder.go

I was :-) but I also wrote:

> This also makes me think about...

So it was unrelated to this suggested change, just saying that we should consider handling alert and socket "printing" as part of the events decoder